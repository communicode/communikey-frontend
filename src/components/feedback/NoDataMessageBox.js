import React from "react";
import PropTypes from "prop-types";
import {Button} from "antd";
import "antd/lib/button/style/index.less";
import "./NoDataMessageBox.less";

/**
 * A message box to signal that there is no data available.
 * It provides an attached call-to-action button.
 *
 * @author sgreb@communicode.de
 * @since 0.8.0
 */
class NoDataMessageBox extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      callToActionButtonDisabled,
      callToActionButtonIconType,
      callToActionButtonVisible,
      headlineText,
      icon,
      onCallToActionButtonClick,
      subHeadlineText
    } = this.props;

    return (
      <div className="container">
        <div className="wrapper">
          <span className="icon">{icon}</span>
          <div className="text">
            <div className="headline">{headlineText}</div>
            <p className="sub-headline">{subHeadlineText}</p>
          </div>
        </div>
        {callToActionButtonVisible &&
          <Button
            className="call-to-action-button"
            icon={callToActionButtonIconType}
            onClick={onCallToActionButtonClick}
            disabled={callToActionButtonDisabled}
        />}
      </div>
    );
  }
}

NoDataMessageBox.propTypes = {
  /**
   * Determines whether the call-to-action button is disabled.
   */
  callToActionButtonDisabled: PropTypes.bool,

  /**
   * The icon type of the call-to-action button.
   */
  callToActionButtonIconType: PropTypes.string,

  /**
   * Determines whether the call-to-action button is visible.
   */
  callToActionButtonVisible: PropTypes.bool,

  /**
   * The text of the headline.
   */
  headlineText: PropTypes.string,

  /**
   * The icon.
   */
  icon: PropTypes.element,

  /**
   * Callback function to handle call-to-action button click events.
   */
  onCallToActionButtonClick: PropTypes.func,

  /**
   * The text of the sub-headline.
   */
  subHeadlineText: PropTypes.string
};

NoDataMessageBox.defaultProps = {
  callToActionButtonDisabled: false,
  callToActionButtonIconType: "plus",
  callToActionButtonVisible: false,
  headlineText: "There is no data yet.",
  icon: null,
  subHeadlineText: ""
};

export default NoDataMessageBox;
