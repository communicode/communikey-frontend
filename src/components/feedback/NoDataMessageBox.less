/*
 * This file is part of communikey.
 * Copyright (C) 2016-2018  communicode AG <communicode.de>
 *
 * communikey is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
.container {
  .wrapper {
    border: 1px solid #efefef;
    border-radius: 4px 4px 0 0;
    padding: 16px;

    .icon {
      font-size: 2.5rem;
      padding-right: 16px;
      vertical-align: middle;
    }

    .text {
      display: inline-block;
      vertical-align: middle;

      .headline {
        font-size: 1rem;
        font-weight: bold;
      }

      .sub-headline {
        font-size: 1rem;
      }
    }
  }

  .call-to-action-button {
    width: 100%;
    border-radius: 0 0 4px 4px;
    margin-top: -1px;
  }
}
