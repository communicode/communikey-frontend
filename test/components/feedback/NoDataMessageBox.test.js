import React from "react";
import { shallow } from "enzyme";
import NoDataMessageBox from "../../../src/components/feedback/NoDataMessageBox";

describe('<NoDataMessageBox>', () => {
  test('should render correctly with sample data', () => {
    const wrapper = shallow(<NoDataMessageBox/>);
    expect(wrapper).toMatchSnapshot();
  });
});