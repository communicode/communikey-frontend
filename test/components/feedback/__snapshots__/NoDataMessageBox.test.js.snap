// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoDataMessageBox> should render correctly with sample data 1`] = `
<div
  className="container"
>
  <div
    className="wrapper"
  >
    <span
      className="icon"
    />
    <div
      className="text"
    >
      <div
        className="headline"
      >
        There is no data yet.
      </div>
      <p
        className="sub-headline"
      />
    </div>
  </div>
</div>
`;
